backend/src/packages/auth/auth.service.ts

```diff
@@ -1,43 +1,51 @@
-import { HttpCode, HttpMessage } from '~/libs/enums/enums.js';
+import { type UserGroupKey, type ValueOf } from 'shared/build';
+import { HttpCode, HttpMessage } from 'shared/build';
+
 import { HttpError } from '~/libs/exceptions/exceptions.js';
 import {
   type CustomerSignUpRequestDto,
-  type CustomerSignUpResponseDto,
+  type UserEntityObjectWithGroupT
 } from '~/packages/users/libs/types/types.js';
 import { type UserService } from '~/packages/users/user.service.js';
 
-import { type UserEntity } from '../users/user.entity.js';
+import { type GroupService } from '../groups/group.service.js';
 
 class AuthService {
   private userService: UserService;
 
-  public constructor(userService: UserService) {
-    this.userService = userService;
-  }
-
-  private async checkExistingUser({
-    phone,
-  }: CustomerSignUpRequestDto): Promise<boolean> {
-    const existingUser: UserEntity | null =
-      await this.userService.findByPhone(phone);
+  private groupService: GroupService;
 
-    return Boolean(existingUser);
+  public constructor(userService: UserService, groupService: GroupService) {
+    this.userService = userService;
+    this.groupService = groupService;
   }
 
-  public async signUp(
-    userRequestDto: CustomerSignUpRequestDto,
-  ): Promise<CustomerSignUpResponseDto> {
-    const isUserExist = await this.checkExistingUser(userRequestDto);
+  public async signUp(groupName: ValueOf<typeof UserGroupKey>,
+    payload: CustomerSignUpRequestDto,
+  ): Promise<UserEntityObjectWithGroupT> {
+    const user = await this.userService.findByEmail(payload.email);
 
-    if (isUserExist) {
+    if (user) {
```

Comment:
Basically I did my own userService from scratch, and then failed to join because after merge with the recent update there were too many discrepancies. I decided to replace this module entirely, and most likely lost the checkIsExistingUser